packages/slo-stat-bot/src/slo-stat-label.ts

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Application, Context} from 'probot';
+import {GitHubAPI} from 'probot/lib/github';
+import {
+  PullsListFilesResponse,
+  PullsListFilesResponseItem,
+  Response,
+  ChecksCreateParams,
+} from '@octokit/rest';
+// eslint-disable-next-line node/no-extraneous-import
+import Ajv, {ErrorObject} from 'ajv';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const schema = require('./../utils/schema.json');
+
+type ValidationResults = {
+  isValid: boolean;
+  errors?: ErrorObject[] | null | undefined;
+};
+
+function handler(app: Application) {
+  app.on(
+    [
+      'pull_request.opened',
+      'pull_request.reopened',
+      'pull_request.edited',
+      'pull_request.synchronize',
+    ],
+    async (context: Context) => {
+      const owner = context.payload.repository.owner.login;
+      const repo = context.payload.repository.name;
+      const pull_number = context.payload.number;
+
+      const fileResponse:
+        | Response<PullsListFilesResponse>
+        | undefined = await handler.listFiles(
+        context.github,
+        owner,
+        repo,
+        pull_number,
+        100
+      );
+      let fileList: PullsListFilesResponseItem[] | undefined;
+
+      if (fileResponse !== undefined) fileList = fileResponse.data;
+
+      for (
+        let i = 0;
+        fileList !== undefined && fileList[i] !== undefined;
+        i++
+      ) {
+        const file = fileList[i];
+
+        if (file.filename === 'issue_slo_rules.json') {
```

Comment:
How does this handle the case where the following files are modified

* ./issue_slo_rules.json
* ./foo/issue_slo_rules.json
* ./.github/issue_slo_rules.json

Looking at it, this would attempt to lint/verify all of them when we would only want to lint `./issue_slo_rules.json` when the repository is named `.github` and we would never want to lint `./foo/issue_slo_rules.json`

Reply:
I will add another test to make sure it does not validate wrong file